Coeat.k step4 pr요청드립니다

kukaro opened the pull request:
안녕하십니까 신입 4기 공채 Coeat입니다.
spring subway리뷰요청드리게되었습니다.
수정된 내역은 아래와 같습니다.

- Section로직을 모두 Line에서 분리하였습니다.
- SectionDao에서 findAll이 Section이 아닌 Sections를 반환하도록 수정하였습니다.
- 깊은 depth의 메소드를 분리하였습니다.

테스트 커버리지는 제가 돌렸을 경우 결과, 아래와 같았습니다.
class/95%, method/87%, line/89%

항상 좋은 피드백 감사드립니다.
새로운 리뷰 부탁드립니다.(아침일찍 대답해주셔서 감사합니다.)
감사합니다

kukaro committed:
refactor : 직관성을 높이기 위해서 LineService의 addSection을 addSectionOnLine으로 변경

kukaro committed:
refactor : addSection을 LineService에서 SectionService로 변경

kukaro committed:
refactor : sections의 생성자를 list<Section>을 받을 수 있도록 변경

kukaro committed:
refactor : line에 관련된 모든 로직 변경 및 분리

kukaro committed:
feat : depth 깊은 메서드 변경

findstar commented on src/main/java/subway/controller/LineController.java:
```diff
@@ -31,34 +32,41 @@ public LineController(LineService lineService,
     public ResponseEntity<LineResponse> updateLine(@PathVariable Long id, @RequestBody LineRequest lineRequest) {
         Line line = lineService.findOneLine(id);
-        LineResponse response = new LineResponse(lineService.updateLine(id, line));
+        Line newLine = lineService.updateLine(line.getId(),
```
새로운 Line 객체를 생성하기 보다 기존의 Line 에서 name 과 color 을 변경하는 메서드를 추가하여
속성값을 변경한 뒤에 이를 다시 updateLine 에 전달하는 방식으로 개선해보면 좋겠습니다.

findstar commented on src/main/java/subway/dao/SectionDao.java:
```diff
@@ -52,17 +53,17 @@ public Section findOne(Long id) {
                 ), id);
```
findOne 메서드는 더 이상 사용되는 곳이 없네요. 
필요하지 않은 메서드는 정리해주세요 😄

findstar commented on src/main/java/subway/domain/Line.java:
```diff
@@ -1,15 +1,11 @@
 
-import java.util.*;
-
 public class Line {
```
Line 객체에서 Sections 가 빠지면서 간결해졌네요. 💯

findstar commented on src/main/java/subway/domain/Sections.java:
```diff
@@ -16,7 +16,13 @@ public Sections() {
     }
 
-    public void initSections(List<Section> sections) {
+    public Sections(Section section){
```
이제 sections 속성값은 다음과 같이 정의하고 인자 없는 생성자는 제거해도 될것 같네요 
```java
private final List<Section> sections = new ArrayList<>();
```

findstar commented on src/main/java/subway/domain/Sections.java:
```diff
@@ -16,7 +16,13 @@ public Sections() {
+        this();
         Map<Station, List<Section>> countMap = new HashMap<>();
         sections.forEach(section -> {
             countMap.computeIfAbsent(section.getUpStation(), (key)->new ArrayList<>());
```
이 객체의 사용하지 않는 메서드도 같이 정리해주세요 

fillSections 에서 스트림 처리 중에 
down station 의 id 를 가져와서 up station 의 id 와 다시 비교하는데요
객체 비교를 할 수는 없을까요?

findstar reviewed (changes requested):
안녕하세요 제임스입니다. 
지난번 피드백 잘 반영해주셨네요. 
Sections 가 좀 더 개선되었네요 💯 
테스트 코드도 잘 수행되는 것 확인했습니다. 
각각의 서비스의 메서드의 역할이 잘 구분되어 있는지 고민해봐주세요. 
추가로 코멘트 몇개 남겼습니다. 확인 부탁드리구요. 
궁금한 점은 언제든지 DM 부탁드립니다.

findstar commented on src/main/java/subway/service/LineService.java:
```diff
@@ -41,8 +41,8 @@ public Line getLine(String name, String color) {
 
-    public void creatSection(Long upStationId, Long downStationId, int distance, Line line) {
-        sectionDao.save(new Section(
+    public Section creatSection(Long upStationId, Long downStationId, int distance, Line line) {
```
Line 이라는 도메인 영역에서 Section 을 알아야 할까요?

kukaro committed:
feat : 피드백 사항 변경

- Line 메소드중에 lineresponse를 받는 메소드 추가
- findOne 메소드를 sectionDao에서 제거
- Sections에서 파라메터 0개의 생성자를 가독성을 위해서 제거
- Sections의 fillSections스트림처리에서 downStation id와 upStation id 비교로직 제거

kukaro commented:
안녕하십니까 신입 공채 4기 Coeat입니다.
피드백 사항을 수정해서 고쳤습니다.
고친 내역은 아래와 같습니다.

- Line 메소드중에 lineresponse를 받는 메소드 추가
- findOne 메소드를 sectionDao에서 제거
- Sections에서 파라메터 0개의 생성자를 가독성을 위해서 제거
- Sections의 fillSections스트림처리에서 downStation id와 upStation id 비교로직 제거

피드백 부탁드립니다. 감사합니다.

findstar commented on src/main/java/subway/controller/LineController.java:
```diff
@@ -31,34 +32,40 @@ public LineController(LineService lineService,
     public ResponseEntity<List<LineResponse>> showAllLines() {
         List<LineResponse> responses = lineService.findAllLines().stream()
-                .map(LineResponse::new)
+                .map(line -> new LineResponse(line,
```
/lines 요청에 대해서는 stations 이 필요하지 않습니다.
LineResponse 를 수정하면 
map(LineResponse::new) 와 같이 좀 더 간단하게 표현할 수도 있을것 같아요.

findstar commented on src/main/java/subway/controller/LineController.java:
```diff
@@ -31,34 +32,40 @@ public LineController(LineService lineService,
                 lineRequest.getName(),
                 lineRequest.getColor());
-        lineService.creatSection(
+        Section section = lineService.creatSection(
```
Section 을 생성하는데 LineService 를 사용하고 있네요  SectionService 로 변경해보면 어떨까요?

findstar reviewed (approved):
안녕하세요. 제임스 입니다. 
지난번 피드백 잘 반영해주셨네요. 
몇가지 코멘트를 남깁니다만 이 PR 은 머지하겠습니다. 
객체를 생성할 때 객체 사이의 관계를 고민해보면 
메서드의 위치를 어디에 해야할지 좀 더 정리할 수 있을것 같습니다.
코드 수정하시느라 수고많으셨습니다
계속 화이팅입니다. 💯
